### DISCUSSION TITLE
Training on Google Collab

### DISCUSSION THREAD

**Discussion by volcacius** [Xilinx/brevitas-radioml-challenge-21#4] [Training]
Hi all,

@louisinhit asked (originally https://github.com/Xilinx/brevitas-radioml-challenge-21/discussions/2#discussioncomment-968626_) if we as organizers provide some remote servers that can be used for training.  We don't, however as an individual user you can easily take advantage of the free GPU resources offered on Google Collab https://colab.research.google.com/ It's a notebook based UX so it should be easy to run your training code there.

Let's keep this discussion open for anything related to training on Google Collab.

Alessandro

**Comment by KristynaPijackova** [comment]
Hi all, 

what are your ways to load the dataset into Colab? 

At the moment I access the dataset via Kaggle [(https://www.kaggle.com/adepitra/radioml2018ota](https://www.kaggle.com/adepitra/radioml2018ota) &  [https://www.kaggle.com/general/74235](https://www.kaggle.com/general/74235) - the dataset link and description of how to easily access it in Colab, if any of you is interested :slightly_smiling_face: ). Also if you have a Kaggle account, you can use the free GPU there as well (only its weekly limited to a certain number of hours).

I used to access the RML2016 datasets through ! wget at first and then just uploaded it on my google drive, which was really comfortable, but the 2018 version is just too big for that :smile: 

If you have any better way to access it, let me know, would love to know! 

Kristyna

**Reply by louisinhit** [reply]
Okay, that's helpful! I will have a look, thanks a lot.

**Reply by maltanar** [reply]
Hi Kristyna, I think you hit the key issue here: Google Drive isn't normally big enough for this unless you have a subscription. An alternative could be to use the local session storage on Google Colab, by opening the left pane, go to Files, then choose "Upload to session storage" to upload RadioML 2018.01A from your local disk. I'm not sure what the limits here are and how long the files will be kept though.

**Reply by KristynaPijackova** [reply]
Hi Yaman, thanks for the reaction :)  To answer your "questions" first, the limit should be around 70Gb, which is more than enough for this dataset. The downside of it is that it is only available for the started session - even if you switch up the CPU and GPU it deletes the files (which is a bit annoying and unfortunate.) 
The approach you described is basically the same as mine with the Kaggle as in both ways the data is stored in the local session storage. But I guess I would not recommend it for most users unless they have a super-fast internet connection. The uploading from the local PC takes me way too long (I'm talking hours) vs. few minutes of downloading from Kaggle (thanks to an internet connection from the server for the download.) But again, thank you very much for the suggestion :)

**Comment by KristynaPijackova** [comment]
Hi all! 

I made a smaller version of the original dataset with 1/8th of the original data to make it easier to work in Google Colab. If anyone is interested in using it, I thought it would be best to provide you with a bit of info and ended up creating a notebook in Colab.

https://colab.research.google.com/drive/1sKSsnv33vbvtpN93Vsa6GYzN_z62Dwtp?usp=sharing

I hoped it is documented enough, so you'd know what you are doing, but basically, you just download the data from my Google Drive (all is included in the notebook) and then just continue with the data loader provided in here (it's just a bit adjusted to the smaller sample sizes). I also added some code snippets of how I got the 1/8th of the data, so it is transparent - did this in Kaggle. And you can also see how you access the dataset in Kaggle (I thought that might come in handy as well). 

**But be careful:**
If you want to pretrain the model on a subset dataset, you'd need to find other way how to extract and store the data. Since I always took the first 512 samples, it might happen that some of my train data might be contained in the test data when you take the full dataset size, thus you might get corrupted accuracy!
I just did what was most comfortable to get me started and did intend to train the whole network from scratch on the full size data. 


If you have any questions, just ask, hope this helps.
